# Test generated by RoostGPT for test roostPostmanTest using AI Type Azure Open AI and AI Model roost-gpt4-32k

# RoostTestHash=6e4c4bbafe

config:
  target: "{{ $processEnvironment.API_HOST }}"
  phases:
    - duration: 60
      arrivalRate: 20
  plugins:
    expect: {}
  payload:
    path: "./order_orderId_delete_others.csv"
    fields:
      - "orderId"
    skipHeader: true

scenarios:
  - name: "Delete purchase order by ID failure, edge and error cases"
    flow:
      - log: "Delete a purchase order with invalid ID"
        post:
          url: "/store/order/9999" # Invalid ID above 1000
          expect:
            - statusCode: 400 # Expect "Invalid ID supplied"
            - hasProperty: "message"
      - think: 1

      - log: "Delete a purchase order with non-integer ID"
        post:
          url: "/store/order/ID123" # Non-integer ID
          expect:
            - statusCode: 400 # Expect "Invalid ID supplied"
            - hasProperty: "message"
      - think: 1

      - log: "Delete a purchase order with missing authorization"
        post:
          url: "/store/order/{{ orderId }}"
          headers:
            Authorization: "InvalidToken" # Invalid authorization token
          expect:
            - statusCode: 401 # Expect "Unauthorized"
            - hasProperty: "message"
      - think: 1

      - log: "Delete a purchase order not found"
        post:
          url: "/store/order/{{ orderId }}"
          headers:
            Authorization: "{{ $processEnvironment.AUTH_TOKEN }}"
          expect:
            - statusCode: 404 # Expect "Order not found"
            - hasProperty: "message"
      - think: 1

      - log: "Delete an order with invalid URL"
        delete:
          url: "/store/invalidURL/order/{{ orderId }}"
          headers:
            Authorization: "{{ $processEnvironment.AUTH_TOKEN }}"
          expect: 
            - statusCode: 404 # Expect page not found as invalid url is passed
            - hasProperty: "message"
